zomes/holo_acorn/code/src/lib.rs

```diff
@@ -223,8 +241,28 @@ mod holo_acorn {
                     validation_package: || {
                         hdk::ValidationPackageDefinition::Entry
                     },
-                    validation: | _validation_data: hdk::LinkValidationData| {
-                        Ok(())
+                   validation: |link_validation_data: hdk::LinkValidationData| {
+                        let validation_data =
+                            match link_validation_data {
+                                hdk::LinkValidationData::LinkAdd {
+                                    validation_data,..
+                                } => validation_data,
+                                hdk::LinkValidationData::LinkRemove {
+                                    validation_data,..
+                                } =>validation_data,
+                            };
+
+                        if let Ok(profile )= hdk::utils::get_as_type::<Profile>(validation_data.package.chain_header.entry_address().clone()){
+                            if profile.address==AGENT_ADDRESS.to_string() {
```

Comment:
Like above, we need to use
`let agent_address = validation_data.package.sources()[0]` instead of `AGENT_ADDRESS`